ebs/error_budget.py

```diff
@@ -417,6 +410,407 @@ def run_etc(self, wfe, wfsc_factor, sensitivity
+        self.eepsr = [config['targets'][star]['eepsr'] 
+                                for star in config['targets']]
+        self.exo_zodi = [config['targets'][star]['exo_zodi'] 
+                                for star in config['targets']]
+        self.exosims_pars_dict = config['initial_exosims']
+        self.write_ppFact_fits()
+        self.exosims_pars_dict['ppFact'] = self.ppFact_filename
+        self.exosims_pars_dict['cherryPickStars'] = self.target_list
+        self.exosims_pars_dict['starlightSuppressionSystems'][0]\
+            ['core_contrast'] = contrast_path
+        self.exosims_pars_dict['starlightSuppressionSystems'][0]\
+            ['core_thruput'] = throughput_path
+        for key in self.exosims_pars_dict['scienceInstruments'][0].keys():
+            if key != 'optics' in dir(self):
+                setattr(self, key
+                        , self.exosims_pars_dict['scienceInstruments'][0][key])
+        for key \
+             in self.exosims_pars_dict['starlightSuppressionSystems'][0] \
+                 .keys():
+            if key in dir(self):
+                setattr(self, key
+                        , self.exosims_pars_dict['starlightSuppressionSystems']
+                                                [0][key])
+
+    def initialize_walkers(self):
+        center = []
+        [center.append(np.array(val).ravel())  
+         for var_name in self.config['mcmc']['variables'] 
+         for val in self.config['mcmc']['variables'][var_name]['ini_pars']
+                               ['center']]
+        center = np.concatenate(center)
+        center = center[np.where(np.isfinite(center))]
+        spread = []
+        [spread.append(np.array(val).ravel())  
+         for var_name in self.config['mcmc']['variables'] 
+         for val in self.config['mcmc']['variables'][var_name]['ini_pars']
+                               ['spread']]
+        spread = np.concatenate(spread)
+        spread= spread[np.where(np.isfinite(spread))]
+        ndim = center.shape[0]
+        nwalkers = self.config['mcmc']['nwalkers']
+        walker_pos = center + np.random.uniform(-spread/2.0, spread/2.0
+                                                , (nwalkers, ndim))
+        return walker_pos
+
+    def update_attributes(self, values):
+        for var_name in self.config['mcmc']['variables']:
+            if var_name in dir(self):
+                template = np.array(self.config['mcmc']['variables'][var_name]
+                                               ['ini_pars']['center'])
+                indices = np.where(np.isfinite(template))
+                use_values, values = np.split(values, [indices[0].size])
+                arr = getattr(self, var_name)
+                if type(arr) == float or type(arr)==np.float64:
+                    arr = use_values[0]
+                else:
+                    arr[indices] = use_values
+                setattr(self, var_name, arr)
```

Comment:
wondering if it would be cleaner to keep track of these values as a dictionary instead of attributes of the ErrorBudgetMcmc class? Especially since the attributes aren't explicitly called anywhere (i.e. `self.QE`) and then I think it should also inherently support adding additional parameters being added without changing the code in the class

Reply:
They became attributes so they can be variables explored by emcee.  If there is a better way to do that, please let me know.